asterius/src/Asterius/Builtins.hs

```diff
@@ -1538,3 +1552,22 @@ trapStoreF64Function _ =
 
 offset_StgTSO_StgStack :: Int
 offset_StgTSO_StgStack = 8 * roundup_bytes_to_words sizeof_StgTSO
+
+
+wrapI64ToI8 _ =
+  let v = runEDSL [] $ do
+        setReturnTypes [I32]
+        x <- param I64
+        -- TODO: add __i64_slot into EDSL.hs
+        storeI64 (symbol "__asterius_i64_slot") 0 x
+        v <- i64Local $ loadI64 (symbol "__asterius_i64_slot") 0
```

Comment:
`loadI8` maybe?

Reply:
Oh god, I'm stupid. Yes, this should be the fix.